pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -27,19 +29,18 @@ import (
 )
 
 func newETCD3Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, error) {
-	tlsInfo := transport.TLSInfo{
-		CertFile: c.CertFile,
-		KeyFile:  c.KeyFile,
-		CAFile:   c.CAFile,
-	}
-	tlsConfig, err := tlsInfo.ClientConfig()
-	if err != nil {
-		return nil, nil, err
-	}
-	// NOTE: Client relies on nil tlsConfig
-	// for non-secure connections, update the implicit variable
-	if len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0 {
-		tlsConfig = nil
+	var tlsConfig *tls.Config
+	if !(len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0) {
```

Comment:
I'm fine reverting the refactoring.
But we need to know why it sets back to nil again.
For example, if a user pass in partial config and it still creates a valid tlsConfig, then it will be nil again. Does it violate the expectation?